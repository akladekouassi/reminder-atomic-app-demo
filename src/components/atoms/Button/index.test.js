import { shallow, render } from "enzyme";

import Button from ".";

describe("Primary Button", () => {
  it("Renders Primary Button", () => {
    const wrapper = render(<Button title="Primary Button" />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
  it("Renders Disabled Button", () => {
    const wrapper = render(<Button title="Disabled Button" disabled />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
  it("Handles button click", () => {
    const mockFunction = jest.fn();
    const wrapper = shallow(
      <Button title="Button Click" onClick={mockFunction} />
    );
    wrapper.simulate("click");
    wrapper.update();
    expect(mockFunction).toBeCalled();
  });
});
