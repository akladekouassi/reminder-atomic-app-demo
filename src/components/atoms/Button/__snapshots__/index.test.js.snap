// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary Button Renders Disabled Button 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled",
      "disabled": "",
      "tabindex": "-1",
      "type": "button",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "MuiButton-label",
        },
        "children": Array [
          Object {
            "data": "Disabled Button",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "button",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "disabled": undefined,
      "tabindex": undefined,
      "type": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "disabled": undefined,
      "tabindex": undefined,
      "type": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Primary Button Renders Primary Button 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary",
      "tabindex": "0",
      "type": "button",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "MuiButton-label",
        },
        "children": Array [
          Object {
            "data": "Primary Button",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "button",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "tabindex": undefined,
      "type": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "tabindex": undefined,
      "type": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
